flight/Modules/Actuator/actuator.c

```diff
@@ -343,29 +370,53 @@ static void actuator_task(void* parameters)
 			clipped = true;
 
 			offset = 1.0f - max_chan;
+		} else if (min_chan < 0.0f) {
+			clipped = true;
+
+			/* Low-side clip management-- how much power are we
+			 * willing to add??? */
+
+			neg_clip /= num_motors;
```

Comment:
it's not num_neg_clip above.  It's the cumulative amount of clip.  What we do here is index it, and figure out how much the throttle has already been increased by clipping.

If we have 4 motors, and two are commanded to -.10 and two to .30.... the user has asked for 10% throttle, and we have sped up two motors 20% and slowed down two motors 20%.

But the actual throttle command achieved by this (assuming linearized actuators) is (0 + 0 + .30 + .30) / 4 = 15%.  So the throttle has already been increased by 5% by clipping.

If maximum throttle advance is 0%, the math below this point isn't quite right-- it's a bit of a fudge-- to lower each motor by 5% still doesn't quite hit the user throttle command-- it takes us to 12.5% in that case.  But really getting that right requires iteratively decreasing the throttle and watching how many motors are in clip, and it's considerably better than what we would do before.

Hm, maybe that can be further improved a little though.  Will think about it.